services/github/github-issues.service.js

```diff
@@ -238,35 +255,98 @@ module.exports = class GithubIssues extends GithubAuthV3Service {
       ? `${isGhLabelMultiWord ? `"${label}"` : label} `
       : ''
-    const labelSuffix = isPRVariant[variant] ? 'pull requests' : 'issues'
+    const labelSuffix = isPR ? 'pull requests' : 'issues'
 
     return {
       label: `${labelPrefix}${labelText}${labelSuffix}`,
-      message: `${metric(numIssues)} ${messageSuffix}`,
-      color: numIssues > 0 ? 'yellow' : 'brightgreen',
+      message: `${metric(issueCount)} ${messageSuffix}`,
+      color: issueCount > 0 ? 'yellow' : 'brightgreen',
     }
   }
 
-  async fetch({ variant, user, repo, label }) {
-    const isPR = isPRVariant[variant]
-    const isClosed = isClosedVariant[variant]
-    const query = `repo:${user}/${repo}${isPR ? ' is:pr' : ' is:issue'}${
-      isClosed ? ' is:closed' : ' is:open'
-    }${label ? ` label:"${label}"` : ''}`
-    const options = { qs: { q: query } }
-    return this._requestJson({
-      url: `/search/issues`,
-      options,
-      schema,
-      errorMessages: errorMessagesFor('repo not found'),
-    })
+  async fetch({ isPR, isClosed, user, repo, label }) {
+    const commonVariables = {
+      user,
+      repo,
+      labels: label ? [label] : undefined,
+    }
+    if (isPR) {
+      const {
+        data: {
+          repository: {
+            pullRequests: { totalCount },
+          },
+        },
+      } = await this._requestGraphql({
+        query: gql`
+          query(
+            $user: String!
+            $repo: String!
+            $states: [PullRequestState!]
+            $labels: [String!]
+          ) {
+            repository(owner: $user, name: $repo) {
+              pullRequests(states: $states, labels: $labels) {
+                totalCount
+              }
+            }
+          }
+        `,
+        variables: {
+          ...commonVariables,
+          states: isClosed ? ['MERGED', 'CLOSED'] : ['OPEN'],
+        },
+        schema: pullRequestCountSchema,
+        transformErrors,
+      })
+      return { issueCount: totalCount }
+    } else {
+      const {
+        data: {
+          repository: {
+            issues: { totalCount },
+          },
+        },
+      } = await this._requestGraphql({
+        query: gql`
+          query(
+            $user: String!
+            $repo: String!
+            $states: [IssueState!]
+            $labels: [String!]
+          ) {
+            repository(owner: $user, name: $repo) {
+              issues(states: $states, labels: $labels) {
+                totalCount
+              }
+            }
+          }
+        `,
+        variables: {
+          ...commonVariables,
+          states: isClosed ? ['CLOSED'] : ['OPEN'],
+        },
+        schema: issueCountSchema,
+        transformErrors,
+      })
+      return { issueCount: totalCount }
+    }
   }
 
   async handle({ variant, raw, user, repo, label }) {
-    const json = await this.fetch({ variant, user, repo, label })
+    const isPR = isPRVariant[variant]
+    const isClosed = isClosedVariant[variant]
```

Comment:
I also moved the logic of converting `variant` to `isPR` and `isClosed` to `handle()`. IMO as this concern is related to interpreting the badge URL, it doesn't really belong to fetching or rendering.

Reply:
Makes sense, this also does the lookup once instead too which is a plus 👍